src/PermissionProvider/ConfigPermissions.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+namespace Drupal\entity_sync\PermissionProvider;
+
+use Drupal\Core\StringTranslation\StringTranslationTrait;
+
+/**
+ * Provides permissions for each entity_sync.sync operation.
+ */
+class ConfigPermissions {
+
+  use StringTranslationTrait;
+
+  /**
+   * Returns an array of permissions for entity_sync.sync configs.
+   *
+   * @return array
+   *   The entity_sync.sync permissions.
+   *   @see \Drupal\user\PermissionHandlerInterface::getPermissions()
+   */
+  public function getPermissions() {
+    $permissions = [];
+
+    $config_factory = \Drupal::configFactory();
+
+    // Loop through each entity_sync.sync configurations and create
+    // corresponding permissions.
+    foreach (
+      $config_factory->listAll('entity_sync.sync.') as $config_name
+    ) {
```

Comment:
**CHANGE REQUEST**

Based on Drupal coding standards [conditions should not be wrapped into multiple lines](https://www.drupal.org/docs/develop/standards/coding-standards#controlstruct).

If it exceeds the line length, do the following:

```
$config_names = $config_factory->listAll('entity_sync.sync.');
foreach ($config_names as $config_name) {
}
```

However, you should use the method provided for fetching multiple config objects at once.

```
$sync_names = $config_factory->listAll('entity_sync.sync.');
foreach ($config_factory->loadMultiple($sync_names) as $sync) {
}
```